### DISCUSSION TITLE
Dependent Tests failing

### DISCUSSION THREAD

**Discussion by paparaju** [avajs/ava#2711] [General]
Hello,
I want to run tests in serial for few testcases and fail the tests if dependent test already failed. Is there any feature available?

**Comment by novemberborn** [comment]
Do you want to fail the tests, or stop running them? For the latter you can use the `--fail-fast` flag or `failFast: true` configuration.

**Reply by paparaju** [reply]
Thank you @novemberborn . This flag is definitely useful. Is it possible to declare any dependencies between tests if they are serial. Basically want to skip a test if any dependent test fails.

**Reply by paparaju** [reply]
I used --fail-fast and it really fails early and indicates that tests are skipped. If the tests are skipped , is it possible to get indication of test failed as part of before and after hooks.

**Reply by novemberborn** [reply]
> Is it possible to declare any dependencies between tests if they are serial. Basically want to skip a test if any dependent test fails.

Not natively within AVA, no.



> If the tests are skipped , is it possible to get indication of test failed as part of before and after hooks.

No, hooks are not run for tests that aren't run.

**Reply by paparaju** [reply]
Thank you for your reply
